### ISSUE TITLE
regex for dockerfile name

### ISSUE THREAD

**Issue by devscheffer** [hadolint/hadolint#1010]
- [x] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
hadolint to work on dockerfiles named dockerfile.test

### Actual behavior
hadolint doesn't find the dockerfiles

### Steps to reproduce the behavior
create dockerfile with name dockerfile.test

rev: v2.12.0

**Comment by romain-dartigues** [comment]
I agree it would be nice for hadolint to work like pyflakes, automatically/recursively find appropriate files.

However, why not follow https://github.com/hadolint/hadolint/blob/v2.12.0/docs/INTEGRATION.md#travis-ci suggestions?

I currently use something along the lines of (in a GitLab context):

```sh
status=0
find -iname '*Dockerfile*' -type f |
while read -r file;
do
  hadolint --no-fail -f gitlab_codeclimate "$file" > "gl-code-quality/$(md5sum "${file}" | head -c32).json";
  hadolint "$file" || status=$(($status|$?));
done
exit "$status"
```
